src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -186,10 +187,41 @@ public void start(final Map<String, String> props) {
         Optional<String> jsonDocument = Optional.empty();
         if (publishFullDocumentOnly) {
           if (changeStreamDocument.containsKey("fullDocument")) {
-            jsonDocument = Optional.of(changeStreamDocument.getDocument("fullDocument").toJson());
+            if (sourceConfig.getJsonType().equals("canonical")) {
```

Comment:
I'd extract all of that at least in another class. There a main function is getting the `changeStreamDocument` and the `sourceConfig` and returning you the `jsonDocument`. The function would check the config (with the `EnumValidator`) and then dispatch to dedicated function.

Reply:
i created a new function for this